processing/src/main/java/org/apache/druid/query/groupby/GroupByQuery.java

```diff
@@ -409,7 +412,10 @@ public boolean getContextSortByDimsFirst()
   @JsonIgnore
   public boolean isApplyLimitPushDown()
   {
-    return applyLimitPushDown;
+    if (forceLimitPushDown == null) {
+      forceLimitPushDown = validateAndGetForceLimitPushDown();
+    }
+    return forceLimitPushDown || canPushDownLimit;
```

Comment:
super nit: it seems funny to have `...LimitPushDown` and `...PushDownLimit`, but I'm not really sure what is the best way to square these up. I guess `...LimitPushDown` is more prevalent, so maybe renaming to `canDoLimitPushDown` to be more consistent?

Reply:
Renamed.